src/tls/openssl.c

```diff
@@ -312,6 +318,101 @@ static int windows_load_system_certificates(struct tls_context *ctx)
 }
 #endif
 
+#ifdef __APPLE__
+/* macOS-specific system certificate loading */
+static int macos_load_system_certificates(struct tls_context *ctx)
+{
+    X509_STORE *store = NULL;
+    X509 *x509 = NULL;
+    SecCertificateRef cert = NULL;
+    CFArrayRef certs = NULL;
+    CFDataRef certData = NULL;
+    const unsigned char *data = NULL;
+    char *subject = NULL;
+    char *issuer = NULL;
+    OSStatus status;
+    unsigned long err;
+    int ret = -1;
+    int loaded_cert_count = 0;
```

Comment:
Do we need to use long type instead of in type here?

In my local development macOS laptop, the following command indicates that there are less than 256 certificates:

```console
$ security find-certificate -a /System/Library/Keychains/SystemRootCertificates.keychain | grep issu | wc -l
     154
```

But some of the enterprises managed macOS laptop could be needed to inject company specific certificates.
What do you think of it?

Reply:
Yep, lets give it a long